hrjobcontract/js/src/job-contract/services/contract-leave.js

```diff
@@ -210,13 +210,14 @@ define([
               return this.leave_amount;
             };
 
-            _.each(leaveType, function (type, typeId) {
+            _.each(leaveType, function (type) {
               modelEntry.leave_type = type.id;
               modelEntry.leave_type_title = type.title;
               modelEntry.leave_calculation_unit_name = type.calculation_unit_name;
               modelEntry.leave_calculation_unit_label = type.calculation_unit_label;
-              modelEntry.leave_amount = 0;
+              modelEntry.leave_amount = parseFloat(type.default_entitlement);
               modelEntry.is_add_public_holidays_readonly = type.calculation_unit_name === 'hours';
+              modelEntry.add_public_holidays = !!parseInt(type.add_public_holiday_to_entitlement);
```

Comment:
i don't think `parseInt` is necessary, you're casting to boolean anyway

Reply:
The `type.add_public_holiday_to_entitlement` is a string containing either 0 or 1.

![image](https://user-images.githubusercontent.com/3973243/33564350-3b79a2ba-d912-11e7-9f0d-ab7c2b707056.png)